Sources/RedisKit/RedisSet.swift

```diff
@@ -1,143 +1,173 @@
-#warning("Re-implement RedisSet")
-//extension RedisDatabase {
-    /// Creates a `RedisSet` for the provided key, that contains the declared type of elements.
+extension RedisClient {
+    /// Creates a `RedisSet` reference to the value stored at `key` with values of `type`.
     ///
-    ///     let idSet = redis.createSetReference(fromKey: "ids", ofType: Int.self)
-    ///     // idSet represents a Set of `Int`.
+    ///     let setOfIDs = client.makeSetReference(key: String, type: Int.self)
+    ///     // setOfIDs represents a Set of `Int`.
     ///
-    /// - Parameter fromKey: The key to identify the Set in this `RedisClient`.
-    /// - Parameter ofType: The type of the elements contained within the set.
-//    public func createSetReference<T>(fromKey key: String, ofType type: T.Type) -> RedisSet<T> {
-//        return RedisSet(identifier: key, using: self)
-//    }
+    /// - Parameters:
+    ///     - key: The Redis key to identify the set.
+    ///     - type: The Swift type representation of the elements in the set.
+    /// - Returns: A `RedisSet` for the key and element type specified.
+    @inlinable
+    public func makeSetReference<T>(key: String, type: T.Type = T.self) -> RedisSet<T> {
```

Comment:
Maybe this could be named `setBuilder` or `makeSetBuilder`?

Reply:
Eh, we're not making a `SetBuilder` that would act as the factory - this method is the factory method itself for making a `RedisSet`.

Perhaps just `makeRedisSet(key:type:)`?